src/xdg.c

```diff
@@ -503,12 +504,10 @@ xdg_toplevel_view_map(struct view *view)
 			position_xdg_toplevel_view(view);
 		}
 
-		if (!view->fullscreen && requested->fullscreen) {
-			set_fullscreen_from_request(view, requested);
-		} else if (!view->maximized && requested->maximized) {
-			view_maximize(view, true,
-				/*store_natural_geometry*/ true);
-		}
+		set_fullscreen_from_request(view, requested);
+		view_maximize(view, requested->maximized ?
+			VIEW_AXIS_BOTH : VIEW_AXIS_NONE,
+			/*store_natural_geometry*/ true);
```

Comment:
This is a logic change, previously we'd do one or the other (or none) and now we do both. I didn't look into the implications of that but thought I should point it out.

Reply:
view_maximize() currently no-ops if fullscreens, I believe.